ts/private/npm_repositories.bzl

```diff
@@ -63,10 +74,35 @@ def _http_archive_version_impl(rctx):
         executable = False,
     )
 
+    if rctx.attr.check_for_updates:
+        vars = ["{}={}".format(v, rctx.os.environ[v]) for v in _TELEMETRY_VARS if v in rctx.os.environ]
+        if rctx.attr.bzlmod:
+            vars.append("bzlmod=true")
+        url = "https://update.aspect.build/aspect_rules_ts/{}?{}".format(
+            RULES_TS_VERSION,
```

Comment:
If someone is consuming from source they'll send `0.0.0` as the RULES_TS_VERSION? How is that handled by `update.aspect.build`?

Reply:
No they would send this `$Format...` string actually https://github.com/aspect-build/rules_ts/blob/main/ts/private/versions.bzl#L9
I should filter that out here instead.

Reply:
The downside is that anyone using a commit SHA rather than a release will just not get counted in our telemetry. I think that's a pretty small fraction of traffic so we can live with that?

Reply:
Can we not send that SHA in that case? I'm not sure if that's accessible anywhere...?

Reply:
I guess we couldn't (easily) check for updates with only a SHA but could still be valuable?

Reply:
We won't know the SHA. It isn't baked into the sources anywhere.

Counting unpinned uses is valuable. Don't think you should filter it out at the service. Unpinned would mean they are bleeding edge from HEAD for some reason or have forked potentially forked it.

Reply:
I was wondering if the SHA from the original http_archive might somehow be available, but I assume not.

We could put the version or commit hash into the source but that might not be worth it?

Reply:
I do think we should send every usage possible though. Let the client send any info available and let the server decide if/what to ignore

Reply:
Okay, I rewrote it as `v0.0.0` and still call the service. On the server-side, we'll have to special-case this and not report that there's an update available.
(I considered making this something like v99.99.99 so it would artificially appear as "later than all releases" but I'd rather not invent something new. We already understand 0.0.0 is the "from HEAD" version in our rulesets.)